### PR TITLE
[plugin_platform_interface] Migrate  to v2 stable, null-safe

### PR DESCRIPTION
**Pull request by kevmoo**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/plugin_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +1,4 @@
-## 1.1.0-nullsafety.2
-
-* Use Mockito null safe.
-
-## 1.1.0-nullsafety.1
-
-* Bump Dart SDK to support null safety.
-
-## 1.1.0-nullsafety
+## 2.0.0
```
Comment (stuartmorgan) [inline packages/plugin_platform_interface/CHANGELOG.md:1]: Why are you deleting old changelog entries?

**Review by stuartmorgan — COMMENTED** [review]
We need to wait on this until we're ready to stable-launch all plugins in the repo, and we'll need tool changes to incrementally block building the plugins we haven't migrated yet, or all our CI will fail.

File: packages/plugin_platform_interface/CHANGELOG.md (hunk 1 above)
Comment (kevmoo) [reply packages/plugin_platform_interface/CHANGELOG.md:1]: They are hidden for most users. In general we don't include pre-release things in the stable changelog

**Comment by stuartmorgan** [comment]
> We need to wait on this until we're ready to stable-launch all plugins in the repo, and we'll need tool changes to incrementally block building the plugins we haven't migrated yet, or all our CI will fail.

Per chat discussion, this isn't actually true; we can do a major-version-spanning range for the dependency since there's no actual breakage between the last version and this one, allowing everything in the repo to still build together.

File: packages/plugin_platform_interface/CHANGELOG.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/plugin_platform_interface/CHANGELOG.md:1]: 👍

**Comment by stuartmorgan** [comment]
If you merge the latest master this should go green.

**Comment by kevmoo** [comment]
> If you merge the latest master this should go green.

Done!

**Review by stuartmorgan — APPROVED** [review]

**Pull request merged**